cli/cmd/utils.go

```diff
@@ -77,82 +83,88 @@ func getConfiguredPlatform() (platform.Platform, hcl.Diagnostics) {
 	return platform, platform.LoadConfig(&lokoConfig.RootConfig.Cluster.Config, lokoConfig.EvalContext)
 }
 
-// getAssetDir extracts the asset path from the cluster configuration.
-// It is empty if there is no cluster defined. An error is returned if the
-// cluster configuration has problems.
-func getAssetDir() (string, error) {
-	cfg, diags := getConfiguredPlatform()
-	if diags.HasErrors() {
-		return "", fmt.Errorf("cannot load config: %s", diags)
-	}
-	if cfg == nil {
-		// No cluster defined and no configuration error
-		return "", nil
+// getKubeconfig finds the right kubeconfig file to use for an action and returns it's content.
+//
+// If platform is required and user do not have it configured, an error is returned.
+func getKubeconfig(contextLogger *logrus.Entry, lokoConfig *config.Config, platformRequired bool) ([]byte, error) {
+	sources, err := getKubeconfigSource(contextLogger, lokoConfig, platformRequired)
+	if err != nil {
+		return nil, fmt.Errorf("selecting kubeconfig source: %w", err)
 	}
 
-	return cfg.Meta().AssetDir, nil
-}
-
-func getKubeconfig() ([]byte, error) {
-	path, err := getKubeconfigPath()
-	if err != nil {
-		return nil, fmt.Errorf("failed getting kubeconfig path: %w", err)
+	// If no sources has been returned, it means we should read from Terraform state.
+	if len(sources) == 0 {
+		return readKubeconfigFromTerraformState(contextLogger)
 	}
 
-	if expandedPath, err := homedir.Expand(path); err == nil {
-		path = expandedPath
+	// Select first non-empty source and read it.
+	for _, source := range sources {
+		if source != "" {
```

Comment:
IIUC, sources will always have path string, so I think there is no need to check if it is empty or not

Reply:
if `KUBECONFIG` environment variable is not set, sources will look like: `[]string{"", "~/.kube/config"}`, so this is needed IMO.

Reply:
Cool, get it